### PR TITLE
Add shadow to divider on BottomSheetDaySessionsFragment 

### LINKED ISSUE #198: Add shadow when scroll sessions
## Kind (Required)

- Improvement

 ## Overview (Required)

- When scroll session, we would like to add shadow to here 📜 
image

 ## Links	

-

### PR DESCRIPTION
**Pull request by rnitame**
## Issue
- close #198 

## Overview (Required)
- Add shadow to divider on BottomSheetDaySessionsFragment when scrolled sessions.
    - Using `RecyclerView.OnScrollListener`

## Links
- Referenced: https://stackoverflow.com/questions/35724308/android-toolbar-elevation-when-scrolling

## Screenshot
demo

### CODE REVIEW

**Comment by yurihondo** [comment]
👀

**Comment by rnitame** [comment]
@yurihondo 
すみません、英語でうまく説明できなさそうなので日本語で失礼します。

今回の #198 ですが、divider の background にアルファ値がついた色を指定しているため、うまく elevation が表示されない問題に遭遇しました。(issue tracker にあがっていて、fix されていません。 https://issuetracker.google.com/issues/37008403 )
手元でアルファ値がない black (#000) で試したところ表示された（下の画像）ため、アルファ値ではなくグレーに近い色で今回の色を表現できると嬉しいのですがいかがでしょうか。

**Comment by yurihondo** [comment]
@rnitame 
Thank you for your helpful comment 👍 
Sounds difficult... 🤔 
We will consider your proposal and contact you again 🙇

**Comment by rnitame** [comment]
@yurihondo 

> We will consider your proposal and contact you again 🙇

Sure, I'll be waiting for you :)
I'm going to work on another issue during waiting :)

**Comment by yurihondo** [comment]
@rnitame 
Sorry for the late reply....
Please wait a little more 🙇

**Comment by yurihondo** [comment]
@rnitame 
Sorry for my too late reply.

I checked this.
I think difficult to add shadow to direct divider as you said.
So, How about add a shadow view like conference app 2019?
https://github.com/DroidKaigi/conference-app-2019/blob/51bb2996a822571923c8562d71b9d999ebf4a341/feature/session/src/main/res/layout/fragment_bottom_sheet_sessions.xml#L116

And control visibility of the shadow at BottomSheetSessionsFragment like so.
https://github.com/DroidKaigi/conference-app-2019/blob/94a9955b319720cacfd28975ba9fb4698f043631/feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/BottomSheetFavoriteSessionsFragment.kt#L97

Could you please consider this solution 🙇 ?

**Comment by rnitame** [comment]
@yurihondo I applyed the solution of conference-app-2019!
Please review 🙏

**Comment by rnitame** [comment]
3 errors ...? 🤔 
image

**Comment by yurihondo** [comment]
👀

**Comment by yurihondo** [comment]
Thank you for your quickly change!

>3 errors ...?
Please wait for confirmation...
Sorry for inconvenience 🙇

File: feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/BottomSheetSessionsFragment.kt [hunk 1]
```diff
@@ -85,6 +86,15 @@ class BottomSheetSessionsFragment : DaggerFragment() {
                 requireContext()
             )
         )
+        binding.sessionRecycler.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+
+                val scrollDirectionUp = -1
```
Comment (yurihondo) [inline feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/BottomSheetSessionsFragment.kt]: Sorry for my nitpick...
Could you please define this variable as a static or put directly into the argument of #canScrollVertically?

Comment (rnitame) [reply feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/BottomSheetSessionsFragment.kt]: Fixed!

**Comment by yurihondo** [comment]
@rnitame 
Thank you for helpful cooperation!!!

I'm terribly sorry to request change you many times...
We found the wrong point for our design of this session list..
There is a feeling of strangeness that there is only a shadow under the divider...

Could you please make the following additional modification however
it goes beyond the scope of the origin issue...?

1. Please change the visibility of divider to `invisible` if list is scrolled. (#canScrollVertically return true)

2. Please change a width of the shadow view to match recyclers view like so.

**Comment by rnitame** [comment]
> Could you please make the following additional modification

OK!

> however it goes beyond the scope of the origin issue... 🙇 ?

I think so, too. I want to merge and create another pull request.

**Comment by yurihondo** [comment]
@rnitame 
Thank you for your reply.

>I think so, too. I want to merge and create another pull request.
Of course.
So I will merge this PR first 👍 

And I created an issue for this.
Could you please any comment like "✋"? 
(I can't assign without a comment... 🙇 )

**Pull request merged**